### PR TITLE
upgrade r2dbc libs (1.0)

### PR DESCRIPTION
**Pull request by pjfanning**
relates to #40

newer r2dbc libs bring in newer netty libs

### CODE REVIEW

File: core/src/main/scala/org/apache/pekko/persistence/r2dbc/internal/R2dbcExecutor.scala [hunk 1]
```diff
@@ -58,12 +58,13 @@ import reactor.core.publisher.Mono
...
   def updateBatchInTx(stmt: Statement)(implicit ec: ExecutionContext): Future[Int] = {
-    val consumer: BiConsumer[Int, Integer] = (acc, elem) => acc + elem.intValue()
+    val consumer: BiConsumer[Long, java.lang.Long] = (acc, elem) => acc + elem.intValue()
     Flux
       .from[Result](stmt.execute())
       .concatMap(_.getRowsUpdated)
-      .collect(() => 0, consumer)
+      .collect(() => 0L, consumer)
       .asFuture()
+      .map(_.intValue())
```
Comment (pjfanning) [inline core/src/main/scala/org/apache/pekko/persistence/r2dbc/internal/R2dbcExecutor.scala:67]: raised https://github.com/apache/incubator-pekko-persistence-r2dbc/issues/90 for this

File: project/Dependencies.scala [hunk 2]
```diff
@@ -32,17 +32,10 @@ object Dependencies {
...
-    // This is here because sbt's ivy resolver doesn't properly support packaging.type when
-    // resolving via sbt-license-report, see https://github.com/sbt/sbt-license-report/issues/87
+    val r2dbcSpi = "io.r2dbc" % "r2dbc-spi" % "1.0.0.RELEASE"
+    val r2dbcPool = "io.r2dbc" % "r2dbc-pool" % "1.0.1.RELEASE"
     val r2dbcPostgres = Seq(
-      ("org.postgresql" % "r2dbc-postgresql" % "0.9.3.RELEASE").excludeAll(
-        "io.netty.incubator", "netty-incubator-codec-native-quic"),
-      ("io.netty.incubator" % "netty-incubator-codec-native-quic" % "0.0.33.Final")
-        .artifacts(Artifact("netty-incubator-codec-native-quic",
-          url("https://repo1.maven.org/maven2/io/netty/incubator/netty-incubator-codec-native-quic/0.0.33.Final/netty-incubator-codec-native-quic-0.0.33.Final.jar"))))
```
Comment (pjfanning) [inline project/Dependencies.scala:45]: this jar no longer seems to be needed

**Comment by pjfanning** [comment]
We have branched 1.0.x so this is safe to merge to main. Would anyone have time to review?

File: core/src/main/scala/org/apache/pekko/persistence/r2dbc/internal/R2dbcExecutor.scala [hunk 3]
```diff
@@ -58,12 +58,13 @@ import reactor.core.publisher.Mono
     }
 
   def updateBatchInTx(stmt: Statement)(implicit ec: ExecutionContext): Future[Int] = {
-    val consumer: BiConsumer[Int, Integer] = (acc, elem) => acc + elem.intValue()
+    val consumer: BiConsumer[Long, java.lang.Long] = (acc, elem) => acc + elem.intValue()
```
Comment (He-Pin) [inline core/src/main/scala/org/apache/pekko/persistence/r2dbc/internal/R2dbcExecutor.scala:61]: there is a ObjLongConsumer

**Review by He-Pin — APPROVED** [review]
lgtm

Hope flux has something like `collectToLong` method:)

File: core/src/main/scala/org/apache/pekko/persistence/r2dbc/internal/R2dbcExecutor.scala (hunk 1 above)
Comment (He-Pin) [inline core/src/main/scala/org/apache/pekko/persistence/r2dbc/internal/R2dbcExecutor.scala:67]: map(_.intvalue)(ExecutionContexts.parasitic)

Comment (He-Pin) [reply core/src/main/scala/org/apache/pekko/persistence/r2dbc/internal/R2dbcExecutor.scala:67]: map(_.intvalue)(ExecutionContexts.parasitic)

Comment (pjfanning) [reply core/src/main/scala/org/apache/pekko/persistence/r2dbc/internal/R2dbcExecutor.scala]: done

**Review by He-Pin — APPROVED** [review]
Lgtm

**Pull request merged**